packages/components/src/FilterBar/FilterBar.component.js

```diff
@@ -180,16 +208,18 @@ export class FilterBarComponent extends React.Component {
 						dockable={this.props.dockable}
 						t={t}
 					/>
-					<Action
-						className={theme.remove}
-						id={this.props.id && `${this.props.id}-cross-icon`}
-						bsStyle="link"
-						icon="talend-cross"
-						label={t('LIST_FILTER_REMOVE', { defaultValue: 'Remove filter' })}
-						hideLabel
-						tooltipPlacement={this.props.tooltipPlacement}
-						onClick={this.props.onToggle}
-					/>
+					{this.state.value ? (
+						<Action
+							className={theme.remove}
+							id={this.props.id && `${this.props.id}-cross-icon`}
+							bsStyle="link"
+							icon="talend-cross"
+							label={t('LIST_FILTER_REMOVE', { defaultValue: 'Remove filter' })}
+							hideLabel
+							tooltipPlacement={this.props.tooltipPlacement}
+							onClick={this.onClear}
```

Comment:
To avoid to play with manual focus, you can use `onMouseDown` and add a `e.preventDefault();`

```suggestion
							onMouseDown={this.onClear}
```